llvm/lib/Support/APFloat.cpp

```diff
@@ -4574,7 +4671,11 @@ IEEEFloat::opStatus IEEEFloat::next(bool nextDown) {
       // the integral bit to 1, and increment the exponent. If we have a
       // denormal always increment since moving denormals and the numbers in the
       // smallest normal binade have the same exponent in our representation.
-      bool WillCrossBinadeBoundary = !isDenormal() && isSignificandAllOnes();
+      // If there are only exponents, any increment always crosses the
+      // BinadeBoundary.
+      bool WillCrossBinadeBoundary =
+          !isDenormal() &&
+          (APFloat::hasSignificand(*semantics) ? isSignificandAllOnes() : true);
```

Comment:
this is an and

Reply:
Sure, simplified this in the latest revision.